go/p2p/rpc/client.go

```diff
@@ -509,8 +508,19 @@ func (c *client) sendRequestAndDecodeResponse(
 }
 
 // NewClient creates a new RPC client for the given protocol.
-func NewClient(p2p P2P, runtimeID common.Namespace, protocolID string, version version.Version, opts ...ClientOption) Client {
-	pid := NewRuntimeProtocolID(runtimeID, protocolID, version)
+func NewClient(p2p P2P, runtimeID *common.Namespace, protocolID string, version version.Version, opts ...ClientOption) Client {
```

Comment:
My plan was to change this when I get to my RPC taks. I would change this to `NewClient(p2p P2P, p core.ProtocolID, ...)` and let the caller decide between `NewProtocolID` and `NewRuntimeProtocolID`. This way you don't need to work with pointers.

I think that the `runtimeID` is never used so this can be removed. Unless I'm missing something.

Reply:
Yeah I think it makes sense to just take a `core.ProtocolID` here and make this up to the caller.

Reply:
Agreed, updated.